internal/validation/validation.go

```diff
@@ -176,8 +176,9 @@ func validateTypeRestrictions(typesys *typesystem.TypeSystem, tk *openfgav1.Tupl
 	return fmt.Errorf("type '%s' is not an allowed type restriction for '%s#%s'", userType, objectType, tk.GetRelation())
 }
 
-// ValidateCondition enforces conditions on a relationship tuple
-func ValidateCondition(typesys *typesystem.TypeSystem, tk *openfgav1.TupleKey) error {
+// validateCondition returns an error if the condition of the tuple is required but not present,
+// or if the tuple provides a condition but it is invalid according to the model.
+func validateCondition(typesys *typesystem.TypeSystem, tk *openfgav1.TupleKey) error {
```

Comment:
🤔 (future suggestion): Perhaps it's time to improve the package design for `typesystem`? It would feel more natural to have something like `type.System` instead IMO.

Reply:
🤷‍♀️ that'd be a huge breaking change, changing the package name

Reply:
We've had some things on our list of things to improve in the Typesystem package, and we have yet to get around to it. But if we're going to do this let's start with what we want to improve and why and propose this as a maintainer team, and then we can evaluate how that fits into our roadmap and the scope of the effort.